.github/actions/happy-build-push-deploy/action.yml

```diff
@@ -0,0 +1,184 @@
+    description: "Specify true to skip migrations"
+    default: "false"
+    required: false
+  working-directory:
+    description: "The happy project root"
+    default: "."
+    required: false
+  name:
+    description: "Name of the built docker image."
+    required: true
+  registry:
+    description: "URI for your docker registry. Typically an ECR registry."
+    required: true
+  build_args:
+    description: "A list of build arguments to pass to the docker build."
+    required: false
+  custom_tag:
+    description: "A single custom tag to apply to the built image."
+    required: false
+    default: ""
+outputs:
+  tags:
+    description: "The tags we built and pushed"
+    value: ${{ steps.meta.outputs.tags }}
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/checkout@v2
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v2
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v2
+    - name: Docker meta
+      id: meta
+      uses: docker/metadata-action@v3
+      with:
+        images: ${{ inputs.registry }}/${{ inputs.name }}
+        tags: |
+          type=ref,event=branch,prefix=branch-
+          type=ref,event=pr,prefix=pr-
+          type=semver,pattern={{version}}
+          type=semver,pattern={{major}}.{{minor}}
+          type=sha,prefix=sha-
+          type=sha,format=long,prefix=sha-
+          ${{ inputs.custom_tag }}
+    - name: Login to ECR
+      uses: docker/login-action@v1
+      with:
+        registry: ${{ inputs.registry }}
+    - name: Calculate Branch and Base Names
+      id: refs
+      uses: chanzuckerberg/github-actions/.github/actions/get-github-ref-names@get-github-ref-names-v1.4.0
+    - name: Calculate Cache-From
+      id: cache-from
+      uses: actions/github-script@v6
+      with:
+        script: |
+          let cacheFrom = [
+            "${{ steps.refs.outputs.baseRef }}",
+            "${{ steps.refs.outputs.headRef }}",
+          ].map(ref => ref.replaceAll(/[^a-zA-Z0-9-_\.]+/g, "-"))
+            .map(ref => `type=registry,ref=${{ inputs.registry }}/${{ inputs.name }}:branch-${ref}`).join('\r\n');
+          console.log(`Will use cached images from ${JSON.stringify(cacheFrom, null, 2)}`);
+          core.setOutput("cacheFrom", cacheFrom);
+    - uses: actions/checkout@v2
+    - name: Install happy
+      uses: chanzuckerberg/github-actions/.github/actions/install-happy@install-happy-v1.4.2
+      with:
+        happy_version: ${{ inputs.happy_version }}
+    - name: Build and push using Happy
+      id: build-push
+      # NOTE: In looking at docker's homebuilt gha for this, it's not clear
+      # to me how exactly they're writign to outputs for `parse-tag` to pick up later
+      env:
+        TFE_TOKEN: ${{ inputs.tfe-token }}
+        STACK_NAME: ${{ inputs.stack-name }}
+        TAGS: ${{ steps.meta.outputs.tags }}
+        LABELS: ${{ steps.meta.outputs.labels }}
+        CACHE-FROM: ${{ steps.cache-from.outputs.cacheFrom }}
```

Comment:
o nice! does this work? If so that would be sick

Reply:
we shall see